components/admin_console/team_channel_settings/team/details/team_members/team_members.tsx

```diff
@@ -125,7 +143,36 @@ export default class TeamMembers extends React.PureComponent<Props, State> {
     }
 
     private search = async (term: string) => {
-        this.props.actions.setSystemUsersSearch(term);
+        this.props.actions.setUserGridSearch(term);
+    }
+
+    private onFilter = async (filterOptions: FilterOptions) => {
```

Comment:
Maybe we can extract this as a util function since its used in multiple components, and it seems to be the same logic?

Reply:
it has slightly different logic one of them grabs roles prefixed with `team`  and one of them grabs `channel` roles technically I could extract a bit of the common logic and add a parameter for 'channel' or 'team' but i didnt think it was worth it, a little bit of duplication here is fine since we may need to modify it for each block differently in the future.